R/recycle.R

```diff
@@ -1,31 +1,125 @@
-#' Vector recycling
+#' Recycling rules used by r-lib and the tidyverse
+#'
+#' @description
```

Comment:
We also talk about these rules in this vignette but I didn't make changes there. I think it is consistent with what I have here https://vctrs.r-lib.org/articles/type-size.html#common-sizes-recycling-rules